proposals/lip-0055.md

```diff
@@ -105,16 +115,42 @@ The full processing of a block is organized as follows.
 5. **Assets verification**: Each module verifies the respective entry in the block assets. If any check fails, the block is discarded and has no further effect.
 6. **Block forwarding**: After the initial checks, the full block is forwarded to a subset of peers.
 7. **Before transactions execution**: Each module can define protocol logic that is executed *before* the transactions contained in the block are processed.
-8. **Transaction verification**: Each module can define protocol logic that verifies a transaction, possibly by accessing the state store. If the verification fails, the transaction is invalid and the whole block is discarded.
-9. **Command verification**: The command corresponding to the `moduleID`-`commandID` combination is verified.
-10. **Before command execution**: Each module can define protocol logic that is processed *before* the command has been executed.
-11. **Command execution**: The command corresponding to the `moduleID`-`commandID` combination is executed.
-12. **After command execution**: Each module can define protocol logic that is processed *after* the command has been executed.
+8. **Transaction verification**: Each module can define protocol logic that verifies a transaction, possibly by accessing the state store. If an error occurs, the transaction is invalid and the whole block is discarded.
+9. **Command verification**: The command corresponding to the `moduleID`-`commandID` combination is verified. If an error occurs, the transaction is invalid and it is not included in the block.
+10. **Before command execution**: Each module can define protocol logic that is processed *before* the command has been executed. If an error occurs, the transaction is invalid and the whole block is discarded.
+11. **Command execution**: The command corresponding to the `moduleID`-`commandID` combination is executed. If an error occurs, the transaction is failed and the processing continues.
+12. **After command execution**: Each module can define protocol logic that is processed *after* the command has been executed. If an error occurs, the transaction is invalid and the whole block is discarded.
 13. **After transactions execution**: Each module can define protocol logic that is executed *after* all the transactions contained in the block have been processed.
 14. **Result verification**: Block header properties, which require accessing the state store *after* all state transitions implied by the block have been executed, are verified.
 15. **Block storage**: The block is persisted into the database.
 16. **Peers notification**: Other peers in the P2P network are notified of the new block.
 
+
+### Block Generation Stages
```

Comment:
Currently, we do not give further details below, what actually happens in the block generation steps.

Reply:
added a sentence and fixed a typo

Reply:
What I meant here is that for the block processing, we describe in detail what happens in the `static validation`, `header verification`, `result verification`, i.e., the validations for every single block header property. I think we should have something similar for the block generation, i.e., describe how each block property is set and whether it happens in the header initialization or header finalization.

Reply:
Added the section `Block header generation`